src/porepy/applications/verification_setups/mandel_biot.py

```diff
@@ -0,0 +1,1569 @@
+            approx_displacement=approx_displacement,
+            approx_flux=approx_flux,
+            approx_force=approx_force,
+            approx_pressure=approx_pressure,
+            error_consolidation_degree=error_consolidation_degree,
+            error_displacement=error_displacement,
+            error_flux=error_flux,
+            error_force=error_force,
+            error_pressure=error_pressure,
+            exact_consolidation_degree=exact_consolidation_degree,
+            exact_displacement=exact_displacement,
+            exact_flux=exact_flux,
+            exact_force=exact_force,
+            exact_pressure=exact_pressure,
+            time=t,
+        )
+
+        return collected_data
+
+
+# -----> Exact solution
+class MandelExactSolution:
+    """Exact solutions to Mandel's problem."""
+
+    def __init__(self, setup):
+        """Constructor of the class."""
+        self.setup = setup
+
+        # For convenience, store the approximated roots
+        self.roots: np.ndarray = self.approximate_roots()
+
+    def approximate_roots(self) -> np.ndarray:
+        """Approximate roots of infinite series using the bisection method.
+
+        Returns:
+            Array of ``shape=(num_roots, )`` containing the approximated roots.
+
+        Note:
+            We solve
+
+            .. math::
+
+               f(x) = \\tan{x} - \\frac{1 - \\nu}{\\nu_u - \\nu} x = 0
+
+            numerically to get all positive solutions to the equation. These are used
+            to compute the infinite series associated with the exact solutions.
+            Experience has shown that 200 roots are enough to achieve accurate results.
+            We find the roots using the bisection method.
+
+            Thanks to Manuel Borregales who helped us with the implementation of this
+            part of the code.
+
+        """
+        # Retrieve physical data
+        nu_s = self.setup.poisson_coefficient()  # [-]
+        nu_u = self.setup.undrained_poisson_coefficient()  # [-]
+
+        # Define algebraic function
+        def f(x):
+            y = np.tan(x) - ((1 - nu_s) / (nu_u - nu_s)) * x
+            return y
+
+        n_series = self.setup.params.get("number_of_roots", 200)
+        a_n = np.zeros(n_series)  # initialize roots
+        x0 = 0.0  # initial point
+        for i in range(n_series):
+            a_n[i] = opt.bisect(
+                f,  # function
+                x0 + np.pi / 4,  # left point
+                x0 + np.pi / 2 - 2.2204e-9,  # right point
+                xtol=1e-30,  # absolute tolerance
+                rtol=1e-14,  # relative tolerance
+            )
+            x0 += np.pi  # apply a phase change of pi to get the next root
+
+        return a_n
+
+    # ---> Pressure
+    def pressure_profile(self, x: np.ndarray, t: number) -> np.ndarray:
+        """Exact fluid pressure in `Pa`.
+
+        Parameters:
+            x: Points in the horizontal axis in meters.
```

Comment:
Yes, here it gets a bit cloudy the documentation. It should be more proper to say _in units of length_ instead of _in meters_.

But then, it gets impractical with, for example _m^2 s^-1_. Let's discuss how to unify documentation of units in person.

Reply:
Fixed as discussed in person.